epyqlib/device.py

```diff
@@ -655,12 +656,15 @@ def _init_from_parameters(
 
         notifiees = []
 
+        # Load the matrix once and copy as needed as a speed optimization.
+        matrix_template = load_matrix(self.can_path)
+
         if Elements.dash in self.elements:
             self.uis = self.dash_uis
 
-            matrix = load_matrix(self.can_path)
             # TODO: this is icky
             if Elements.tx not in self.elements:
+                matrix = copy.copy(matrix_template)
```

Comment:
If a shallow `copy.copy()` is sufficient, maybe nothing is needed?  Any modifications to the `list` or `dict` etc attributes would affect all copies anyways.  It would only protect against assignments directly to the top level class instance's attributes.

https://github.com/ebroecker/canmatrix/blob/490a7833b64a554095b38bc47584cad6b3b02ceb/src/canmatrix/canmatrix.py#L1522

It isn't helpful at the top level, but just since I ran into it, canmatrix does have a copy module.  (or maybe you already saw this)

https://github.com/ebroecker/canmatrix/blob/490a7833b64a554095b38bc47584cad6b3b02ceb/src/canmatrix/copy.py.